lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -463,6 +468,31 @@ static Value splitBOperand(Value b, mlir::PatternRewriter &rewriter) {
   return newB;
 }
 
+struct DescriptorLoadAttr {
+  bool fp4Padded;
+  bool twoCTA;
+  unsigned swizzleBytes;
+};
+
+void propageteDescriptorLoadAttributes(Value operand,
+                                       const DescriptorLoadAttr &attrs) {
+  SetVector<Operation *> backwardSlice;
+  BackwardSliceOptions opt;
+  getBackwardSlice(operand, &backwardSlice, opt);
+
+  for (auto op : backwardSlice) {
+    if (auto descLoad = dyn_cast<ExperimentalDescriptorLoadOp>(op)) {
+      auto newDescLoadAttr = mlir::triton::DescriptorLoadAttr::get(
+          op->getContext(), attrs.fp4Padded, attrs.twoCTA);
+      descLoad.setDescAttrAttr(newDescLoadAttr);
+    } else if (auto makeDesc = dyn_cast<MakeTensorDescOp>(op)) {
+      auto newMakeDescAttr = mlir::triton::MakeDescriptorAttr::get(
+          op->getContext(), attrs.fp4Padded, attrs.swizzleBytes);
+      makeDesc.setDescAttrAttr(newMakeDescAttr);
+    }
+  }
+}
```

Comment:
there are multiple problems with propagating this way. First the slice may lead you to the wrong op if going through ops like scf.if/scf.for and second if the slice doesn't fine the matching ops we will silently miscompile.